자동 로그인

knh0113 opened the pull request.

knh0113 committed:
자동 로그인

knh0113 committed:
로그인, 회원가입 초안

donsonioc2010 commented on Api/src/main/java/picasso/server/api/user/controller/AuthController.java:
```diff
@@ -28,9 +31,29 @@ public class AuthController {
+                if (cookie.getName().equals("userId")) {
+                    String userId = cookie.getValue();
+                    Optional<User> findResult = userService.findUserById(Long.parseLong(userId));
+                    findResult.ifPresent(user -> setSessionLoginUser(session, user));
```
코드로 봤을때 기능은 될 거 같은데.. 그러면 사용자가 임의로 userId라는 Key로 1, 2 이런식으로 넣거나 변경할 경우 다른 사용자정보로 로그인 되도록 Injection해볼 수 있지 않을까요?

그나마 좀 더 안전하게 ID, Email 이런식으로 두개, 세개의 정보를 넣고 가입시키는게 좋지 않을까 하네요

donsonioc2010 commented on Api/src/main/java/picasso/server/api/user/controller/AuthController.java:
```diff
@@ -42,16 +65,63 @@ public String showLoginForm(HttpSession session) {
+                    break;
+                }
+            }
+        }
```
자동로그인 힘줘서 쿠키에 저장 다했는데 로그아웃때  삭제를 할 경우에는,
다음에 로그인시 쿠키를 통한 자동로그인이 가능할까요 못할까요?

한번 생각해보시면 좋을 것 같습니다.

knh0113 committed:
자동 로그인기능, 로그인 회원가입 화면 초안

donsonioc2010 commented on Api/src/main/java/picasso/server/api/user/controller/AuthController.java:
```diff
@@ -42,16 +65,63 @@ public String showLoginForm(HttpSession session) {
+        findResult.ifPresent(user -> {
+            setSessionLoginUser(session, user);
+            // 자동 로그인 쿠키 생성
+            if (requestDto.isRememberMe()) {
+                Cookie cookie = new Cookie("userId", user.getId().toString());
+                cookie.setMaxAge(10); // 10초 유지
+                response.addCookie(cookie);
+            }
```
쿠키를 10초만유지하나요?

knh0113 committed:
Merge branch 'dev' into feat/user

donsonioc2010 committed:
fix : 검증문제 해결

donsonioc2010 committed:
Merge branch 'feat/user' of https://github.com/donsonioc2010/picasso into feat/user

knh0113 committed:
Merge branch 'dev' into feat/user

# Conflicts:
#	Api/src/main/java/picasso/server/api/user/service/UserService.java

donsonioc2010 commented on Api/src/main/resources/templates/auth/login.html:
```diff
@@ -4,25 +4,34 @@
-        <button type="submit">로그인</button>
-    </form>
+
+    <div style="margin:80px; background-color:white; display: flex; flex-direction: column; border-radius : 70px; justify-content: center; align-items: center; height:50vh; margin-left: 37%; border: 10px solid #18ACDB; width: 380px;">
```
이건..너무기네요

ID태그 활용하면서 CSS파일 따로 만들어서 사용하면 좋을 것 같습니다.

donsonioc2010 commented on Api/src/main/resources/templates/auth/logout.html:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>$Title$</title>
+</head>
+<body>
+$END$
+</body>
+</html>
```
기능상문제없으면 상관은 없습니다만 ㅋㅋ...

이건 사실 비동기 통신(Ajax)를 공부하고 구현하는게 더 편할거에요.

donsonioc2010 commented on Api/src/main/resources/templates/auth/signup-form.html:
```diff
@@ -4,32 +4,78 @@
 
-<h1>회원가입</h1>
-<form th:action="@{/auth/signup}" method="post">
+  <div style="margin:80px; background-color:white; display: flex; flex-direction: column; border-radius : 70px; justify-content: center; align-items: center; height:50vh; margin-left: 37%; border: 10px solid #18ACDB; width: 380px;">
```
로그인과 마찬가지로 CSS파일 한개를 따로 만들어서 작성하는게 좋을 듯 합니다.

donsonioc2010 commented on Api/src/main/resources/templates/auth/signup-form.html:
```diff
@@ -4,32 +4,78 @@
+
+
+
+<!--
+<!DOCTYPE html>
+<html
+        lang="ko"
+        xmlns="http://www.w3.org/1999/xhtml"
+        xmlns:th="http://www.thymeleaf.org"
+        xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+        layout:decorate="~{layout/layout}"
+>
+<main layout:fragment="content">
+
+  <h1>회원가입</h1>
+  <form th:action="@{/auth/signup}" method="post">
+    <table>
+      <tr>
+        <td><input type="nickname" name="nickName" placeholder="별명"></td>
+      </tr>
+      <tr>
+        <td><input type="email" name="email" placeholder="이메일"></td>
+      </tr>
+      <tr>
+        <td><input type="password" name="password" placeholder="Password"></td>
+      </tr>
+      <tr>
+        <td><input type="password" name="password" placeholder="Password Confirm"></td>
+      </tr>
+    </table>
+    <button type="submit">등록</button>
+  </form>
   <th:block layout:fragment="script">
     <script type="application/javascript" th:inline="javascript">
     </script>
     <script type="text/javascript">
     </script>
   </th:block>
 </html>
+-->
```
불필요하면 삭제해도 괜찮을 것 같아요.

donsonioc2010 reviewed (commented):
확인하였습니다.

knh0113 committed:
Merge branch 'feat/user' of https://github.com/donsonioc2010/picasso into feat/user

knh0113 committed:
회원가입시 패스워드 수정

knh0113 committed:
자동로그인 화면 수정

knh0113 committed:
자동로그인 수정

knh0113 committed:
자동 로그인 수정2

knh0113 committed:
자동 로그인 수정3

knh0113 committed:
feat : Exception 추가

knh0113 committed:
checkout을 위한 Commit

knh0113 committed:
Merge dev Branch into

knh0113 committed:
사용자 수정

knh0113 committed:
Merge branch 'dev' into feat/user

knh0113 committed:
Merge branch 'dev' into feat/user

donsonioc2010 committed:
feat : 입찰기능 추가 (#71)

* refactor, feat : 컬럼 추가, 타입변경 및 rename 에 따른 기능 수정

* feat : Scheduling완료, Mail Template 페이지제작

* tempSave

* feat : 입찰 기능 구현

* feat : checkout을 위한 Commit

* feat : Emial Template 제작중

* feat : 이메일 구현
- 입찰자가 없는 경우 유찰 안내메일
- 관리자가 승인하지 않은 경우 유찰 안내메일
- 경매 시작 안내 메일
- 링크 발송을 위한 Properties 추가

* Issue 수정

* feat : detail페이지 이쁘게 꾸미기

* faet : Front로직 추가~

* Feat : 입찰기능 추가

knh0113 committed:
로그인 화면 수정

knh0113 committed:
로그인 화면 수정

knh0113 committed:
사용자 수정2

knh0113 committed:
사용자 수정2

knh0113 committed:
Merge branch 'dev' into feat/user

# Conflicts:
#	Api/src/main/java/picasso/server/api/user/controller/AuthController.java
#	Api/src/main/java/picasso/server/api/user/service/UserService.java
#	Common/src/main/java/picasso/server/common/exception/GlobalException.java

knh0113 committed:
사용자 수정3

knh0113 committed:
merge

knh0113 committed:
merge 2
